index.html

```diff
@@ -654,5 +654,400 @@ <h2>Exposing change of MediaStreamTrack configuration</h2>
+    <section>
+      <h3>{{HumanFaceLandmarkDetectionMode}}</h3>
+<pre class="idl">
+enum HumanFaceLandmarkDetectionMode {
+  "none",
+  "center-point",
+};</pre>
+      <section class="notoc">
+        <h4>{{HumanFaceLandmarkDetectionMode}} Enumeration Description</h4>
+        <dl data-dfn-for="HumanFaceLandmarkDetectionMode" data-link-for="HumanFaceLandmarkDetectionMode">
+          <dt><dfn><code>none</code></dfn></dt>
+          <dd>
+            <p>This source does not offer object detection for human face landmarks.
+            For setting, this is interpreted as a command to turn off
+            the setting of human face landmark detection metadata.</p>
+          </dd>
+          <dt><dfn><code>center-point</code></dfn></dt>
+          <dd>
+            <p>This source offers detection of the center point of human face landmarks, or such a mode is requested.</p>
+          </dd>
+        </dl>
+      </section>
+    </section>
+    <section>
+      <h3>Constrainable Properties</h3>
+      <ol>
+        <li>
+          <p><dfn>Face detection mode</dfn> describes which face properties
+          are to be detected and set in the metadata for the video frame.</p>
+        </li>
+        <li>
+          <p><dfn>Face landmark detection mode</dfn> describes which face landmark
+          properties are to be detected and set in the metadata for the video frame.</p>
+        </li>
+      </ol>
+    </section>
+    <section>
+      <h3>Examples</h3>
+      <pre class="example">
+// main.js:
+// Check if face detection is supported by the browser
+const supports = navigator.mediaDevices.getSupportedConstraints();
+if (supports.humanFaceDetectionMode) {
+  // Browser supports face contour detection.
+} else {
+  throw('Face contour detection is not supported');
+}
+
+// Open camera with face detection enabled
+const stream = await navigator.mediaDevices.getUserMedia({
+  video: { humanFaceDetectionMode: 'bounding-box' }
+});
+const [videoTrack] = stream.getVideoTracks();
+
+// Use a video worker and show to user.
+const videoElement = document.querySelector('video');
+const videoWorker = new Worker('video-worker.js');
+videoWorker.postMessage({track: videoTrack}, [videoTrack]);
+const {data} = await new Promise(r => videoWorker.onmessage);
+videoElement.srcObject = new MediaStream([data.videoTrack]);
+
+// video-worker.js:
+self.onmessage = async ({data: {track}}) => {
+  const generator = new VideoTrackGenerator();
+  parent.postMessage({videoTrack: generator.track}, [generator.track]);
+  const {readable} = new MediaStreamTrackProcessor({track});
+  const transformer = new TransformStream({
+    async transform(frame, controller) {
+      for (const face of frame.metadata().humanFaces || []) {
+        if (face.boundingBox) {
+          console.log(
+            `Face @ (${face.boundingBox.left}, ${face.face.boundingBox.top}), ` +
+                   `(${face.boundingBox.right}, ${face.face.boundingBox.bottom})`);
+        }
+      }
```

Comment:
If the check I suggested above is added, then this can be simplified:
```suggestion
      for (const face of frame.metadata().humanFaces) {
        console.log(
          `Face @ (${face.boundingBox.left}, ${face.face.boundingBox.top}), ` +
                 `(${face.boundingBox.right}, ${face.face.boundingBox.bottom})`);
      }
```

Reply:
You suggest here two changes:
(1) assume that metadata().humanFaces always exists (so it would be empty array if no faces were detected) and
(2) for each face, assume that face.boundingBox exists because its creation is enabled in the settings

Regarding both (1) and (2):
The feedback we received and the way the metadata is implemented allows users to copy-modify and amend the metadata. Your suggested changes make assumptions which might not hold if the metadata is modified. In this example the assumptions are valid but it might cause bugs later on if the code is reused in another context. I'm mostly happy to merge these changes into the example but I think it's best to add at least a comment *why* the assumptions are made in this case (ie. metadata coming directly from the video track).

Regarding (1):
The change you suggest assumes that if there are no faces then metadata().humanFaces does exist (but is an empty array). It was not `required` in the spec proposal, so I don't see that it would be necessarily true. Maybe we should mark it as `required`?

[metadata().humanFaces was replaced with metadata().segment due to [issue 79](https://github.com/w3c/mediacapture-extensions/issues/79) but generally the same comments hold.]